types/framework/entities/actor.d.ts

```diff
@@ -294,47 +294,56 @@ declare class Actor<
   /** @override */
   createEmbeddedEntity(
     embeddedName: string,
-    data: Optional<ActiveEffect | Actor.OwnedItem<I>>,
+    data: Optional<ActiveEffect.Data | Actor.OwnedItem<I>>,
     options?: any
-  ): Promise<ActiveEffect | Actor.OwnedItem<I>>;
+  ): Promise<ActiveEffect.Data | Actor.OwnedItem<I>>;
```

Comment:
Can we make what is returned more specific with generics, if it depends on what you put in?

Reply:
Yes, either by generics, or by overloading on the `embeddedName`, which can only be `'ActiveEffect'` or `'OwnedItem'`. I used the latter variant because this one also works for `deleteEmbeddedEntity`, so it's more consistent.